File: solutions/fortran/armstrong-numbers/1/armstrong_numbers.f90

module armstrong_numbers
  implicit none
contains

  logical function isArmstrongNumber(i)
    integer, intent(in) :: i
    integer :: k, m, l, n, sum
    if (i==0) then
      isArmstrongNumber = .true.
    else
      n=i
      m=7
      l=m
      sum=0
      do k=1,l
      if (n/10**(7-k)==0) then
          m=m-1
      end if
      sum=sum+(n/10**(7-k))**m
      n=mod(n,10**(7-k))
      end do
      
      isArmstrongNumber = (sum==i)
    end if
  end function

end module

File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>

bool is_armstrong_number(int candidate){
    int c=1,d=0;
    int sum=0;
    int count=0;
    int digits=0;
    c=candidate;
    while(c!=0)
    {
        c=c/10;
        count++;
    }
    if (count==0){
        count=1;
    }
    digits=count;
    count=0;
    c=candidate;
    while(c!=0)
    {
        d=c%10;
        d=pow(d,digits);
        sum=sum+d;
        c=c/10;
        count++;
    }
    if (sum==candidate)
    {
        return true;    
    }
    else
    {
        return false;
    }
}
